### PR TITLE
[publish][m #32]: Upload README

### PR DESCRIPTION
**Pull request by kittdotech**
Upload README file to S3

### CODE REVIEW

File: tests/test_publish_success.py [hunk 1]
```diff
@@ -72,10 +73,11 @@ def test_publish_success(self):
 
         # WHEN `dpm publish` is invoked
         result = self.invoke(cli, ['publish', '--publisher', 'testpub'])
-
         # THEN 'publish ok' should be printed to stdout
         self.assertRegexpMatches(result.output, 'publish ok')
-        # AND 5 requests should be sent
+        #Checking README
+        self.assertIn('Uploading README', result.output)
```
Comment (zelima) [inline tests/test_publish_success.py]: We should also have tests for cases 
- When there is no README at all
- When README does not have extension at all (should be uploaded)
- When README has wrong name Eg: READMEE or README. (not sure about last one)
- When readme is lowercase

**Comment by kittdotech** [comment]
In DPMPY publish flow, I have added Readme to upload to S3:
1) Get Token
2) Upload Datapackage.json
3) Upload Resources
4) Upload README[.*] if exist else show warning message.
    a) Search all README.* in current DP
    b) Get PUT URL to store in S3 (Similar to resource)
    c) Upload the first README found
5) Finalize

@rgrp @zelima Please review

**Comment by zelima** [comment]
@kittdotech I still can't see test for 
- uploading `README` (without extensions)
- Not uploading with wrong names E.g. `README.` or lowercased

Also Why do we have `test_miscellaneous.py ` in separate file?

**Comment by kittdotech** [comment]
Added Test for Readme & Test in Readme Category.

**Comment by zelima** [comment]
@kittdotech Looks good!
Could we include one more test: Check theres uploaded only one README if there is two or more valid README in directory.

**Pull request merged**